frontend/src/concepts/connectionTypes/ConnectionTypeForm.tsx

```diff
@@ -0,0 +1,130 @@
+      return [
+        {
+          value: '',
+          content: connectionType.metadata.annotations['openshift.io/display-name'],
+          isSelected: true,
+        },
+      ];
+    }
+    if (!isPreview && connectionTypes) {
+      return connectionTypes.map((t) => ({
+        value: getResourceNameFromK8sResource(t),
+        content: getDisplayNameFromK8sResource(t),
+        isSelected: t.metadata.name === connectionType?.metadata.name,
+      }));
+    }
+    return [];
+  }, [isPreview, connectionType?.metadata, connectionTypes]);
+
+  return (
+    <Form>
+      {isPreview && <Title headingLevel="h1">Add connection</Title>}
+      <FormGroup label="Connection type" fieldId="connection-type" isRequired>
+        <TypeaheadSelect
+          id="connection-type"
+          selectOptions={options}
+          onSelect={(_, selection) =>
+            setConnectionType?.(connectionTypes?.find((c) => c.metadata.name === selection))
+          }
+          isDisabled={isPreview || connectionTypes?.length === 1}
+          placeholder={
+            isPreview && !connectionType?.metadata.annotations?.['openshift.io/display-name']
+              ? 'Unspecified'
+              : 'Select a type, or search by keyword or category'
+          }
+          toggleProps={
+            isPreview && !connectionType?.metadata.annotations?.['openshift.io/display-name']
+              ? { status: MenuToggleStatus.danger }
+              : undefined
+          }
+        />
+        {connectionType && (
+          <ConnectionTypeDetailsHelperText connectionType={connectionType} isPreview={isPreview} />
+        )}
+      </FormGroup>
+      {(isPreview || connectionType?.metadata.name) && (
+        <FormSection title="Connection details" style={{ marginTop: 0 }}>
+          <K8sNameDescriptionField
+            dataTestId="connection-name-desc"
+            nameLabel="Connection name"
+            descriptionLabel="Connection description"
+            data={
+              connectionNameDesc ?? {
+                name: '',
+                description: '',
+                k8sName: {
+                  value: '',
+                  state: {
+                    immutable: true,
+                    invalidCharacters: false,
+                    invalidLength: false,
+                    maxLength: 0,
+                    touched: false,
+                  },
+                },
+              }
+            }
+            onDataChange={setConnectionNameDesc}
+          />
+          <ConnectionTypeFormFields
+            fields={connectionType?.data?.fields}
+            isPreview={isPreview}
+            connectionValues={connectionValues}
+            onChange={(field, value) => {
+              setConnectionValues?.({
+                ...connectionValues,
+                [field.envVar]: value,
+              });
+            }}
+          />
+        </FormSection>
```

Comment:
Ending the `FormSection` after `ConnectionTypeFormFields` seems to give a lot of extra spacing. Perhaps it should end before?
![Uploading image.png…]()

Reply:
I think this is actually the Section row that is giving the spacing. If there is no section, there is no extra spacing

Reply:
in this screenshot the 1st row is a field, and second is a Section

![image](https://github.com/user-attachments/assets/32fb44cd-82a0-43ef-806f-1d8c4bb31f54)

Reply:
Ok. It looked odd but I confirmed with PF samples and that spacing is expected when a new section starts.